crawl-ref/source/skills.cc

```diff
@@ -1068,14 +1090,31 @@ static int _train(skill_type exsk, int &max_exp, bool simu)
     return skill_inc;
 }
 
-void set_skill_level(skill_type skill, double amount)
+/**
+ * Calculate the difference in skill points and xp for new skill level `amount`.
+ *
+ * If `crosstraining`, `religion`, and/or `manuals` are set, this will produce
+ * an estimate only. Otherwise, it is exact. This function is used in both
+ * estimating skill targets, and in actually changing skills. It will work for
+ * both cases where the new skill level is greater, or less than the current
+ * level.
+ *
+ * @param skill the skill to calculate for.
+ * @param amount the new skill level for `skill`.
+ * @param scaled_training how to scale the training values (for estimating what happens when other skills are being trained).
+ * @param crosstraining whether to factor in crosstraining.
+ * @param religion whether to factor in god bonuses (specifically, ash).
+ * @param manuals whether to factor in manuals.  Does not use exact skill points from manual -- only uses training rate bonus.
+ *
+ * @return a pair consisting of the difference in skill points, and the difference in xp.
+ */
+skill_diff skill_level_to_diffs(skill_type skill, double amount,
+                            int scaled_training,
+                            bool crosstraining, bool religion, bool manuals)
```

Comment:
This is a lot of bools, which will make it difficult to know what is happening at a call site: `skill_level_to_diffs(sk, 5, 10, true, false, true)` is pretty inscrutable. Is there any way to combine them, or use an argument whose value has more meaning? A `mode` enum, or fields in a bitmask or something?